git-branchless-lib/src/core/check_out.rs

```diff
@@ -71,20 +71,36 @@ pub fn check_out_commit(
         render_smartlog,
     } = options;
 
-    let target = match target {
-        None => None,
+    let (target, oid) = match target {
+        None => (None, None),
         Some(CheckoutTarget::Reference(reference_name)) => {
             let categorized_target = CategorizedReferenceName::new(&reference_name);
-            Some(categorized_target.remove_prefix()?)
+            (Some(categorized_target.remove_prefix()?), None)
         }
-        Some(CheckoutTarget::Oid(oid)) => Some(oid.to_string()),
-        Some(CheckoutTarget::Unknown(target)) => Some(target),
+        Some(CheckoutTarget::Oid(oid)) => (Some(oid.to_string()), Some(oid)),
+        Some(CheckoutTarget::Unknown(target)) => (Some(target), None),
     };
 
     if get_undo_create_snapshots(repo)? {
         create_snapshot(effects, git_run_info, repo, event_log_db, event_tx_id)?;
     }
 
+    let target = if let Some(oid) = oid {
+        let all_branches = repo.get_all_local_branches()?;
+        let matching_branches: Vec<&Branch> = all_branches
+            .iter()
+            .map(|b| (b, b.get_oid().unwrap().unwrap()))
+            .filter(|t| t.1 == oid)
+            .map(|t| t.0)
+            .collect();
+        match matching_branches.len() {
+            1 => Some(String::from(matching_branches.get(0).unwrap().get_name()?)),
+            _ => target,
+        }
```

Comment:
You can do a cool pattern-match to avoid the `unwrap` here, something like this:

```suggestion
        match matching_branches.as_slice() {
            &[branch] => Some(branch.get_name()?.to_owned()),
            _ => target,
        }
```

For consistency with the rest of the codebase, use `to_owned()` here to convert the `&str` into a `String`.

Reply:
This code has changed a bit. It's replacement also uses unwrap, but I think it's far less unwieldy this time